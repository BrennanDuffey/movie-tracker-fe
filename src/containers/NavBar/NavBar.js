import React from 'react';
import {NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const NavBar =(props)=> {

  let {name} = props.user;
  let userIcon;
  let userLink;

  if (name) {
    userIcon = 'fas signin-icon fa-user-minus';
    userLink = '/signout';
  } else {
    userIcon = 'fas signin-icon fa-user-plus';
    userLink = '/login';
  }

  return (
    <nav className="NavBar">
        <article className="user--content">
        <NavLink 
          to={userLink}
          onClick={props.handleClick} className="NavBar-Signout">
          <i className={userIcon}></i>
          {!name && <h4>Please Signin</h4>}
          {name && <h4>Hi, {name}</h4>}
        </NavLink>  
        </article>
      <section className="NavButton--container">
        <NavLink exact to="/" className="NavBar-links">In Theaters</NavLink>
        <NavLink to="/favorites" className="NavBar-links">Favorites</NavLink>
        <button onClick={()=> window.alert('Feature Coming Soon')}>Latest Movies</button>
        <button onClick={()=> window.alert('Feature Coming Soon')}>History</button>
        <button onClick={()=> window.alert('Feature Coming Soon')}>Watch List</button>
      </section>
    </nav>
  )
}

NavBar.propTypes={
  dispatch: PropTypes.func,
  user: PropTypes.object
}


export const mapStateToProps =(state)=>({
  user: state.user
});


export default connect(mapStateToProps)(NavBar);