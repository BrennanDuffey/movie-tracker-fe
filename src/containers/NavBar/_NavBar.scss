nav {
  box-sizing: border-box;
  height: 100vh;
  width: 100%;
  background: #970000;
  border: 5px solid rgba(157, 157, 157, 0.4);
  display: grid;
  grid-template-rows: 20% 80%;
  grid-template-areas: "User" "Links";
}

.NavBar-Signout {
  grid-area: User;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding-top: 10%;

  
  i {
    content: '';
    width: 40px;
    height: 40px;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #500000;
    border-radius: 50%;
    border: silver ridge 3px;

    &:hover {
      color: #3a3a3a;
      text-shadow: white 1px 1px 3px;
    }
  }
  
  h4 {
    display: inline;
    margin: 5px;
    color: white;
    
    &:hover {
      color: #3a3a3a;
      text-shadow: #500000 .5px .5px 1px;
    }
  }
}

.signin-icon {
  font-size: 24px;
  color: silver;
  text-shadow: 2px 4px 3px rgba(0, 0, 0, 0.5);
}



.NavButton--container {
  grid-area: Links;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-content: center;

   button, .NavBar-link {
     height: 50px;
     color: white;
     background: none;
     border: none;
     text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.55);
   }

   .active {
     background: #500000;
     border: 4px solid rgba(0, 0, 0, 0.25);
     box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
     border-radius: 0px 20px 20px 0px;
   }

   a {
     text-align: center;
     line-height: 50px;
     height: 50px;
     text-decoration: white none;
     color: white;
     background: none;
     border: none;
     text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.55);
   }
}