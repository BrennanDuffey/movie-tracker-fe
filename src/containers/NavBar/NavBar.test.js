import {NavBar} from './NavBar';
import React from 'react';
import {shallow} from 'enzyme';

let mockUser = {name: 'Abc', id: 1}
let mockNoUser = {name: '', id: 0}

describe('NavBar', ()=>{
  let wrapper;

  it('should match snapshot, w/ user signed in', ()=> {
    wrapper= shallow(<NavBar user={mockUser} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should match snapshot, w/ user signed out', ()=> {
    wrapper= shallow(<NavBar user={mockNoUser} />);
    expect(wrapper).toMatchSnapshot();
  });
});