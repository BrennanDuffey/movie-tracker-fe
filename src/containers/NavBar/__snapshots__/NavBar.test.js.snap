// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot, w/ user signed in 1`] = `
<nav
  className="NavBar"
>
  <article
    className="user--content"
  >
    <NavLink
      className="NavBar-Signout"
      to="/signout"
    >
      <i
        className="fas signin-icon fa-user-minus"
      />
      <h4>
        Hi, 
        Abc
      </h4>
    </NavLink>
  </article>
  <section
    className="NavButton--container"
  >
    <NavLink
      className="NavBar-links"
      exact={true}
      to="/"
    >
      In Theaters
    </NavLink>
    <NavLink
      className="NavBar-links"
      to="/favorites"
    >
      Favorites
    </NavLink>
    <button
      onClick={[Function]}
    >
      Latest Movies
    </button>
    <button
      onClick={[Function]}
    >
      History
    </button>
    <button
      onClick={[Function]}
    >
      Watch List
    </button>
  </section>
</nav>
`;

exports[`NavBar should match snapshot, w/ user signed out 1`] = `
<nav
  className="NavBar"
>
  <article
    className="user--content"
  >
    <NavLink
      className="NavBar-Signout"
      to="/login"
    >
      <i
        className="fas signin-icon fa-user-plus"
      />
      <h4>
        Please Signin
      </h4>
    </NavLink>
  </article>
  <section
    className="NavButton--container"
  >
    <NavLink
      className="NavBar-links"
      exact={true}
      to="/"
    >
      In Theaters
    </NavLink>
    <NavLink
      className="NavBar-links"
      to="/favorites"
    >
      Favorites
    </NavLink>
    <button
      onClick={[Function]}
    >
      Latest Movies
    </button>
    <button
      onClick={[Function]}
    >
      History
    </button>
    <button
      onClick={[Function]}
    >
      Watch List
    </button>
  </section>
</nav>
`;
